engine/src/main/java/org/terasology/rendering/nui/widgets/UIText.java

```diff
@@ -82,6 +82,9 @@
     @LayoutConfig
     protected boolean readOnly;
 
+    @LayoutConfig
+    private boolean passwordMode;
```

Comment:
Yes considered that, but avoided for
> too much duplication

exactly this reason. Can move to this approach if you prefer, though.